File: scripts/create_test_user.go
package main

import (
	"fmt"
	"log"
	"socks5-app/internal/auth"
	"socks5-app/internal/config"
	"socks5-app/internal/database"
)

func main() {
	// 初始化配置
	if err := config.Init(); err != nil {
		log.Fatalf("配置初始化失败: %v", err)
	}

	// 初始化数据库
	if err := database.Init(); err != nil {
		log.Fatalf("数据库初始化失败: %v", err)
	}

	// 生成密码哈希
	password := "123456"
	hashedPassword, err := auth.HashPassword(password)
	if err != nil {
		log.Fatalf("密码加密失败: %v", err)
	}

	fmt.Printf("密码: %s\n", password)
	fmt.Printf("哈希: %s\n", hashedPassword)

	// 创建测试用户
	user := &database.User{
		Username: "testuser3",
		Password: hashedPassword,
		Role:     "user",
		Status:   "active",
	}

	// 检查用户是否已存在
	var existingUser database.User
	if err := database.DB.Where("username = ?", user.Username).First(&existingUser).Error; err == nil {
		// 用户已存在，更新密码
		existingUser.Password = hashedPassword
		if err := database.DB.Save(&existingUser).Error; err != nil {
			log.Fatalf("更新用户失败: %v", err)
		}
		fmt.Printf("用户 %s 的密码已更新\n", user.Username)
	} else {
		// 创建新用户
		if err := database.DB.Create(user).Error; err != nil {
			log.Fatalf("创建用户失败: %v", err)
		}
		fmt.Printf("用户 %s 创建成功\n", user.Username)
	}

	fmt.Println("测试用户信息:")
	fmt.Printf("用户名: %s\n", user.Username)
	fmt.Printf("密码: %s\n", password)
	fmt.Printf("角色: %s\n", user.Role)
}

File: scripts/create_test_user.py
#!/usr/bin/env python3
"""
创建测试用户
"""
import bcrypt
import pymysql

# 数据库配置
DB_CONFIG = {
    'host': '127.0.0.1',
    'port': 3306,
    'user': 'socks5_user',
    'password': 'socks5_password',
    'database': 'socks5_db'
}

def hash_password(password):
    """密码加密"""
    salt = bcrypt.gensalt()
    hashed = bcrypt.hashpw(password.encode('utf-8'), salt)
    return hashed.decode('utf-8')

def create_user(username, password):
    """创建用户"""
    try:
        conn = pymysql.connect(**DB_CONFIG)
        cursor = conn.cursor()
        
        # 检查用户是否存在
        cursor.execute("SELECT id FROM users WHERE username = %s", (username,))
        if cursor.fetchone():
            print(f"用户 {username} 已存在，删除后重建...")
            cursor.execute("DELETE FROM users WHERE username = %s", (username,))
        
        # 创建新用户
        hashed_pwd = hash_password(password)
        sql = """
        INSERT INTO users (username, password, role, status, bandwidth_limit, created_at, updated_at)
        VALUES (%s, %s, 'user', 'active', 0, NOW(), NOW())
        """
        cursor.execute(sql, (username, hashed_pwd))
        conn.commit()
        
        print(f"✓ 用户 {username} 创建成功")
        print(f"  - 用户名: {username}")
        print(f"  - 密码: {password}")
        print(f"  - 角色: user")
        print(f"  - 状态: active")
        print(f"  - 带宽限制: 0 (无限制)")
        
        cursor.close()
        conn.close()
        return True
    except Exception as e:
        print(f"✗ 创建用户失败: {e}")
        return False

if __name__ == "__main__":
    create_user("fwy1014", "fwy1014")
